torchhd/datasets/collection_datasets.py

```diff
@@ -0,0 +1,403 @@
+import os
+import os.path
+from typing import Callable, Optional, Tuple, List
+import torch
+from torch.utils import data
+import pandas as pd
+import tarfile
+import numpy as np
+
+from .utils import download_file_from_google_drive
+
+
+class CollectionDataset(data.Dataset):
+    """`Generic class for loading datasets used in `Do we Need Hundreds of Classifiers to Solve Real World Classification Problems? <https://jmlr.org/papers/v15/delgado14a.html>`_.
+
+    Args:
+        dataset (string): Name of the dataset. Used when extracting the corresponding files from archive.
+        root (string): Root directory containing the files of the dataset.
+        has_test (bool): Indicates whether the dataset has a separate test file or not.
+        download (bool): If True, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+    """
+
+    def __init__(
+        self,
+        dataset: str,
+        root: str,
+        has_test: bool,
+        download: bool,
+    ):
+        self.__dataset = dataset
+        self.__has_test = has_test
+        root = os.path.join(root, self.__dataset)
+        root = os.path.expanduser(root)
+        self.root = root
+        os.makedirs(self.root, exist_ok=True)
+
+        if download:
+            self.download()
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found or corrupted. You can try to specify download=True to download it"
+            )
+
+    def _check_integrity(self) -> bool:
+        if not os.path.isdir(self.root):
+            return False
+
+        # Check if the root directory contains the required files
+        if self.__has_test:
```

Comment:
You can extract this `__has_test` case only to the `DatasetTrainTest` class

Reply:
We use it in the parental class in "_check_integrity" so I think it might still be useful to have it